### ISSUE TITLE
could not load module "8188eu.ko": No such file or directory

### ISSUE THREAD

**Issue by CraftedPvP** [mfruba/kernel#4]
I downloaded the repo and went to the first file you uploaded the one for 4.9.0 and did the commands like so
```
make
make install
insmod /lib/modules/4.9.0-kali3-amd64/kernel/drivers/net/wireless/8188eu.ko
```
but on the third command I can't load the modules

insmod: ERROR: could not load module /lib/modules/4.9.0-kali3-amd64/kernel/drivers/net/wireless/8188eu.ko: No such file or directory

I went to the file directory and I saw nothing was there except some files and folders.
![problem](https://user-images.githubusercontent.com/20311462/27385381-113b4816-56c5-11e7-9e0d-0286828064c5.PNG)
I could see "8188eu.ko" where I saved the repo in. Should I run the command there instead?

**Comment by mfruba** [comment]
Hi, 
Yes you can load it. 
'insmod /lib/modules/4.9.0-kali3-amd64/kernel/drivers/net/wireless/8188eu.ko' does not work because you have probably slightly different version of kernel so just use path with proper kernel version.

**Comment by CraftedPvP** [comment]
![problem](https://user-images.githubusercontent.com/20311462/27415301-e20f965e-5738-11e7-8de0-8b4bd41b0ccc.PNG)
I'm fairly new to Linux version. Can I ask which one is the kernel version? the one on the left or the one on the right?

**Comment by CraftedPvP** [comment]
I did the 3rd command again and now it said
`Couldn't load module. File exist`
What should I do to fix it?

**Comment by mfruba** [comment]
You have to unload previous driver
Run: rmmod 8188eu

**Comment by CraftedPvP** [comment]
well that's also a problem since it reverts back to the title's error problem.
![problem](https://user-images.githubusercontent.com/20311462/27505064-24b503b8-58c9-11e7-9882-920091e99a2d.PNG)

**Comment by mfruba** [comment]
Probably this will solve your problem:
modprobe cfg80211

If not then provide output from dmesg

**Comment by CraftedPvP** [comment]
I did `modprobe` and the 3rd command to install. It printed out no message. Curious, I plugged back my usb adapter and it didn't work. I did dmesg and I can see a lot of errors on my usb adapter.

`dmesg`
![problem](https://user-images.githubusercontent.com/20311462/27513078-f303bbd4-598b-11e7-930a-93b599fd1293.PNG)
![problem1](https://user-images.githubusercontent.com/20311462/27513079-f307bb12-598b-11e7-9c1c-ef67ccc27b70.PNG)

**Comment by Gucci1-wp** [comment]
Hi CraftedPvP

Did you get your TP-Link wireless adapter to work ? I'm experiencing the same issue as you...

Please let me know

**Comment by CraftedPvP** [comment]
Hi Gucci1-wp unfortunately I've asked multiple people and I did some research and well you won't be able to get it to work. Another thing is the one who posted the repo is not responding anymore. Either he doesn't know how to fix our problem or he's busy. It's best if we get a new adapter

**Comment by ViezeVingertjes** [comment]
Works flawlessly on the raspberry, so should be able to replicate that with a regular linux right?

**Comment by Rajssss** [comment]
@ViezeVingertjes is monitor mode is working?

**Comment by alpha-gj** [comment]
I have the same issue that CraftedPvP posted.

**Comment by alpha-gj** [comment]
Hi All,
 
I find this trick. 
1. Please check *.ko by using "`file 8188eu.ko`" command to check its' kernel version is the same as your OS kernel version?

2. The wifi dongle should need to insert it first, and type` sudo insmod 8188.ko`. That will work.

**Comment by ViezeVingertjes** [comment]
@Rajssss Yes on the raspberry (raspbian) i had it all working within 5 minutes, but now i am trying to reproduce this on a 'normal' debian and no succes so far. Might be because i am using Virtualbox issue though, not sure.

I used the auto installer for the raspberry from https://gist.github.com/kmonsoor/59c47d05d8c6cb397c15cd5eba7909e8 (instructions are on top of the file).

**Comment by Ruflead** [comment]
@ViezeVingertjes  how you do that..??

Are you completely enable monitor mode on Tl wn722n v2.
???

**Comment by Torub** [comment]
> modprobe cfg80211

root@kali:~# cd rtl8188eus/
root@kali:~/rtl8188eus# ls
8188eu.ko     8188eu.o                dkms-install.sh  Kconfig         os_dep
8188eu.mod    BUILD_FOR_NETHUNTER.md  dkms-remove.sh   Makefile        platform
8188eu.mod.c  core                    hal              modules.order   README.md
8188eu.mod.o  dkms.conf               include          Module.symvers  ReleaseNotes.pdf
root@kali:~/rtl8188eus# insmod /lib/modules/5.3.0-kali2-amd64/kernel/drivers/net/wireless/8188eu.ko
insmod: ERROR: could not insert module /lib/modules/5.3.0-kali2-amd64/kernel/drivers/net/wireless/8188eu.ko: Unknown symbol in module
root@kali:~/rtl8188eus# module
bash: module: command not found
root@kali:~/rtl8188eus# dmesg | tail -n 1
[ 2999.735119] 8188eu: Unknown symbol cfg80211_mgmt_tx_status (err -2)
root@kali:~/rtl8188eus# modprobe cfg80211
root@kali:~/rtl8188eus# insmod /lib/modules/5.3.0-kali2-amd64/kernel/drivers/net/wireless/8188eu.ko
root@kali:~/rtl8188eus# 
root@kali:~/rtl8188eus#

is it succesful nothing shows

**Comment by MrRa88it** [comment]
i think kernal version not support to Driver (out of date)

**Comment by haxor-dot** [comment]
I was having kinda same issue with my TP-LINK WN722N adapter. Was just freaking about why it's not working. Tried many tutorials but none worked for me. People sometimes put something like "r8188eu" or "8188eu" in blacklist in "etc/modprobe.d/realtek.conf" while following a tutorial on how to enable monitor mode. So I whitelisted the driver by just deleting the line which contains "blacklist r8188eu" or "blacklist 8188eu" and after a reboot it was working. The adapter showed up in the menu.

**Comment by slm9n** [comment]
> I was having kinda same issue with my TP-LINK WN722N adapter. Was just freaking about why it's not working. Tried many tutorials but none worked for me. People sometimes put something like "r8188eu" or "8188eu" in blacklist in "etc/modprobe.d/realtek.conf" while following a tutorial on how to enable monitor mode. So I whitelisted the driver by just deleting the line which contains "blacklist r8188eu" or "blacklist 8188eu" and after a reboot it was working. The adapter showed up in the menu.

Hello 
How did you blacklist the two Drivers?

**Comment by haxor-dot** [comment]
> How did you blacklist the two Drivers?

I did not black list both of them I said you can blacklist the driver which you want to (or have to)
